src/apiutil.h

```diff
@@ -308,3 +316,208 @@ bool hasInsufficientMaterial(Color c, const Position& pos) {
+      curRankWidth += c - '0';
+      // if previous digit exists we have a dual digit
+      if (isdigit(prevChar))
+        curRankWidth += 9;
+    }
+    else if (c == '/') {
+      ++nbProcessedRanks;
+      if (curRankWidth != nbFiles) {
+        std::cerr << "curRankWidth != nbFiles: " << curRankWidth << " != " << nbFiles << std::endl;
+        return FEN_FILE_SIZE_MISMATCH;
+     }
+      curRankWidth = 0;
+    }
+    else if (c == '+') {
+        // ignore '+' symbols in fen (e.g. kyotoshogi_variant)
+    }
+    else  // normal piece
+      ++curRankWidth;
+    prevChar = c;
+  }
+  if (v->pieceDrops) { // pockets can either be defined by [] or /
+    if (nbProcessedRanks != nbRanks && nbProcessedRanks-1 != nbRanks) {
+      std::cerr << "Invalid number of ranks. Expected: " << nbRanks << " Actual: " << nbProcessedRanks << std::endl;
+      return FEN_INVALID_NUMBER_OF_RANKS;
+    }
+  }
+  else {
+    if (nbProcessedRanks != nbRanks) {
+      std::cerr << "Invalid number of ranks. Expected: " << nbRanks << " Actual: " << nbProcessedRanks << std::endl;
+      return FEN_INVALID_NUMBER_OF_RANKS;
+    }
+  }
+
+  // check for pocket
+  if (v->pieceDrops) {
+    if (nbProcessedRanks-1 == nbRanks) {
+      // look for last '/'
+      for (auto it = fenParts[0].rbegin(); it != fenParts[0].rend(); ++it) {
+        if (*it == '/')
+          break;
+        if (*it != '-') {
+          if (v->pieceToChar.find(*it) == std::string::npos) {
+            std::cerr << "Invalid pocket piece: '" << *it << "'." << std::endl;
+            return FEN_INVALID_POCKET_INFO;
+          }
+        }
+      }
+    }
+    else {  // pocket is defined as [ and ]
+      if (*(fenParts[0].end()-1) != ']') {
+        std::cerr << "Pocket specification does not end with ']'." << std::endl;
+        return FEN_INVALID_POCKET_INFO;
+      }
+      // look for first '['
+      for (auto it = fenParts[0].rbegin()+1; it != fenParts[0].rend(); ++it) {
+        if (*it == '[')
+          break;
+        if (*it != '-')
+          if (v->pieceToChar.find(*it) == std::string::npos) {
+              std::cerr << "Invalid pocket piece: '" << *it << "'." << std::endl;
+            return FEN_INVALID_POCKET_INFO;
+          }
+      }
+    }
+  }
+
+  // 2) Part
+  // check side to move char
+  if (fenParts[1][0] != 'w' && fenParts[1][0] != 'b') {
+    std::cerr << "Invalid side to move specification: '" << fenParts[1][0] << "'." << std::endl;
+    return FEN_INVALID_SIDE_TO_MOVE;
+  }
+
+  // 3) Part
+  // check castling rights (exclude seirawan_variant because it has different castling specification)
+  if (v->castling && !v->seirawanGating) {
+    for (char c : fenParts[2])
+      if (c != '-')
+        if (c != 'k' && c != 'K' && c != 'Q' && c != 'q') {
+          std::cerr << "Invalid castling specification: '" << c << "'." << std::endl;
+          return FEN_INVALID_CASTLING_INFO;
+        }
+  }
+
+  // 4) Part
+  // check en-passant square
+  if (v->castling) {
```

Comment:
This should refer to en passant.

Reply:
Is there a boolean to check if the variant has an en passant square?

Reply:
Good point 🙂 . I was not really aware that there is no config to disable en passant explicitly without disabling double steps. However, `doubleStep` is a strict requirement for en passant to occur, so that could be used.

Reply:
Using `doubleStep` as an identifier instead, runs into the problem for the **breakthrough_variant**.
Here the fen is defined as `pppppppp/pppppppp/8/8/8/8/PPPPPPPP/PPPPPPPP w 0 1`, `firstRankDoubleSteps=false` and `doubleStep=true`.
Is this a bug and should `doubleStep` be set to `false` instead?

Reply:
Setting those values in the absence of pawns does not have any effect, therefore this value might be sloppily defined for some variants. However, in the specific case of breakthrough this really looks like a typo (or rather wrong auto-completion), because if I was too lazy to add doubleStep I would not have added the other one for sure... If we want to fully cover the conditions we could in addition to doubleStep check that pawns are in the set of pieceTypes. I leave it up to you if you prefer to make the condition more specific or make the variant definitions more consistent (or both).